src/screens/charging-stations/connector-details/ChargingStationConnectorDetails.tsx

```diff
@@ -727,27 +743,93 @@ export default class ChargingStationConnectorDetails extends BaseAutoRefreshScre
           <View style={style.noButtonStopTransaction} />
         )}
         {/* Details */}
-        <ScrollView
-          contentContainerStyle={style.scrollViewContainer}
-          refreshControl={<RefreshControl refreshing={this.state.refreshing} onRefresh={this.manualRefresh} />}>
-          <View style={style.rowContainer}>
-            {this.renderConnectorStatus(style)}
-            {this.renderUserInfo(style)}
-          </View>
-          <View style={style.rowContainer}>
-            {this.renderInstantPower(style)}
-            {this.renderTotalConsumption(style)}
-          </View>
-          <View style={style.rowContainer}>
-            {this.renderElapsedTime(style)}
-            {this.renderInactivity(style)}
-          </View>
-          <View style={style.rowContainer}>
-            {this.renderBatteryLevel(style)}
-            {isPricingActive ? this.renderPrice(style) : <View style={style.columnContainer} />}
-          </View>
-        </ScrollView>
+        {connector?.status === ChargePointStatus.AVAILABLE ? (
+          <ScrollView
+            contentContainerStyle={style.scrollViewContainer}
+            refreshControl={openUserModal ? null : <RefreshControl refreshing={this.state.refreshing} onRefresh={this.manualRefresh} />}>
+            <View style={style.rowContainer}>{this.renderConnectorStatus(style)}</View>
+            {isAdmin && this.renderUserSelection(style, formStyle)}
+          </ScrollView>
+        ) : (
+          <ScrollView
+            contentContainerStyle={style.scrollViewContainer}
+            refreshControl={<RefreshControl refreshing={this.state.refreshing} onRefresh={this.manualRefresh} />}>
+            <View style={style.rowContainer}>
+              {this.renderConnectorStatus(style)}
+              {this.renderUserInfo(style)}
+            </View>
+            <View style={style.rowContainer}>
+              {this.renderInstantPower(style)}
+              {this.renderTotalConsumption(style)}
+            </View>
+            <View style={style.rowContainer}>
+              {this.renderElapsedTime(style)}
+              {this.renderInactivity(style)}
+            </View>
+            <View style={style.rowContainer}>
+              {this.renderBatteryLevel(style)}
+              {isPricingActive ? this.renderPrice(style) : <View style={style.columnContainer} />}
+            </View>
+          </ScrollView>
+        )}
       </Container>
     );
   }
+
+  private openUserModal(open: boolean): void {
```

Comment:
Whether you use switchDisplayUserModal() and you invert the flag or you create two method, one that opens and one that closes the modal.
But for one statement, no need to have a method.